libcxx/test/libcxx/utilities/variant/variant.variant/variant_size.pass.cpp

```diff
@@ -84,5 +100,8 @@ int main(int, char**) {
   static_assert(sizeof(std::variant<char, int, long>) == sizeof(type_with_index<long>));
   static_assert(sizeof(std::variant<std::size_t, std::size_t, std::size_t>) == sizeof(type_with_index<std::size_t>));
 
+  static_assert((sizeof(VariantWithNoUniqueAddress) < sizeof(VariantWithoutNoUniqueAddress)) ==
+                ExpectSmallerSizeWithNoUniqueAddress);
+
```

Comment:
I think we should ensure that the following still works - i.e. `b` won't be accidentally overwritten.
At least we should add this to the test suite (into `libcxx/test/std` subdirectory).

```C++
#include <cassert>
#include <variant>

#include "test_macros.h"

struct S {
  [[no_unique_address]] std::variant<int, void*> a;
  bool b;
};

TEST_CONSTEXPR_CXX20 bool test() {
  S x{{}, true};
  x.a.emplace<0>();
  x.a.emplace<1>();
  return x.b;
}

#if TEST_STD_VER >= 20
static_assert(test());
#endif

int main() {
  assert(test());
}
```

Reply:
Done.